packages/orchestration/src/orchestration-client.ts

```diff
@@ -79,15 +79,31 @@ export class OrchestrationClient {
       );
     }
 
-    return this.createStreamResponse(
-      {
-        prompt,
-        requestConfig,
-        stream: true,
-        streamOptions: options
-      },
-      controller
-    );
+    try {
+      return await this.createStreamResponse(
+        {
+          prompt,
+          requestConfig,
+          stream: true,
+          streamOptions: options
+        },
+        controller
+      );
+    } catch (error: any) {
+      if (
+        error.cause?.response?.data &&
+        typeof error.cause.response.data.read === 'function'
+      ) {
+        const chunks: Buffer[] = [];
+        for await (const chunk of error.cause.response.data) {
+          chunks.push(Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk));
+        }
+        const responseString = Buffer.concat(chunks).toString('utf8');
+        const data = JSON.parse(responseString);
+        error.cause.response.data = data;
+      }
```

Comment:
[req] You need to put this part in `http-client.ts` in the core package. Because there `ErrorWithCause` is instantiated and `addStack()` was called. Changing the `error.cause.response.data` afterwards won't affect the stack anymore.

So in the catch block of `executeRequest()`, do this

```ts
    if (
      error.response?.data &&
      typeof error.response.data.read === 'function'
    ) {
      const chunks: Buffer[] = [];
      for await (const chunk of error.response.data) {
        chunks.push(Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk));
      }
      error.response.data = JSON.parse(Buffer.concat(chunks).toString('utf8'));
    }
```